### PR TITLE
fix: checks for status in employee and for date range in upload attendance.

### PR DESCRIPTION
**Pull request by sahil28297**
1. Earlier the status of an amployee can be set to 'Left' even though other employees are reporting to this employee, added checks to prevent this behaviour.
2. Earlier the range of dates added to the template in upload attendance were not checked for date of joining and relieving date. Now the range of dates added in the template are between the date of joining and relieving date.
image

### CODE REVIEW

**Review by netchampfaris — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/employee/employee.py [hunk 1]
```diff
@@ -147,8 +147,21 @@ def validate_email(self):
...
-		if self.status == 'Left' and not self.relieving_date:
-			throw(_("Please enter relieving date."))
+		if self.status == 'Left':
+			reports_to = frappe.db.get_all('Employee',
+				fields=['name', 'reports_to'],
+				filters={'reports_to': self.name}
+			)
+
+			if reports_to:
+				employee_names = []
```
Comment (Alchez) [inline erpnext/hr/doctype/employee/employee.py]: Use list comprehension for this as well.

**Review by Alchez — CHANGES_REQUESTED** [review]

**Review by Alchez — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/employee/employee.py [hunk 2]
```diff
@@ -147,8 +147,18 @@ def validate_email(self):
...
-		if self.status == 'Left' and not self.relieving_date:
-			throw(_("Please enter relieving date."))
+		if self.status == 'Left':
+			reports_to = frappe.db.get_all('Employee',
+				fields=['name', 'reports_to'],
+				filters={'reports_to': self.name}
+			)
+			if reports_to:
+				employee_names = [row.name for row in reports_to]
+				link_to_employees = [frappe.utils.get_link_to_form('Employee', employee_name) for employee_name in employee_names]
```
[the remark this answers is not in the document]
Comment (sahil28297) [reply erpnext/hr/doctype/employee/employee.py]: will get complicated from readability point of view

[the remark this answers is not in the document]
Comment (Alchez) [reply erpnext/hr/doctype/employee/employee.py]: `link_to_employees = [frappe.utils.get_link_to_form('Employee', employee.name) for employee in reports_to]`?

[the remark this answers is not in the document]
Comment (sahil28297) [reply erpnext/hr/doctype/employee/employee.py]: yeah, this works.

**Review by netchampfaris — APPROVED** [review]

**Pull request merged**